core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -578,8 +585,12 @@ impl<S: ReadStorage, Tr: Tracer + Default + 'static> VmInterface for Vm<S, Tr> {
         let start = self.inner.world_diff().snapshot();
         let gas_before = self.gas_remaining();
 
-        let mut full_tracer = (mem::take(tracer), CircuitsTracer::default());
+        let mut full_tracer = (
+            (mem::take(tracer), CircuitsTracer::default()),
+            EvmDeployTracer::new(self.world.dynamic_bytecodes.clone()),
+        );
         let result = self.run(execution_mode, &mut full_tracer, track_refunds);
+        let (full_tracer, _) = full_tracer;
         *tracer = full_tracer.0; // place the tracer back
```

Comment:
Doing this in two steps is weird. Either fully pattern match or do `full_tracer.0.0`.